### PR TITLE
microerror: handle nil in Stack func

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: funcs.go [hunk 1]
```diff
@@ -15,6 +15,10 @@ import (
 //	logger.LogCtx(ctx, "level", "error", "message", "failed to do a thing", "stack", microerror.Stack(err))
 //
 func Stack(err error) string {
+	if err == nil {
+		return "<nil>"
```
Comment (xh3b4sd) [inline funcs.go]: I think zero value would be better here. Any reason not to go with it?

Comment (kopiczko) [reply funcs.go]: I think when you see json like `"stack": "<nil>"` it is more reassuring that this isn't a bug in the printing code.

Comment (kopiczko) [reply funcs.go]: And `<nil>` is taken from `fmt.Printf("%v\n", nil)` output  https://play.golang.org/p/h-4E8grakeS

Comment (xh3b4sd) [reply funcs.go]: Ok then we could just add the nil case below. 

```
case nil:
	fmt.Printf("%v", nil)
```

Comment (kopiczko) [reply funcs.go]: It won't work because this is type assertion switch but I can do:
```suggestion
		return fmt.Printf("%v", nil)
```

Comment (xh3b4sd) [reply funcs.go]: It works. I tried it out.

Comment (kopiczko) [reply funcs.go]: TIL. Changed. I also added test.

Comment (xh3b4sd) [reply funcs.go]: I didn't knew that either but I was curious and saw an opportunity that then validated. 🎉

**Pull request merged**